go/test/endtoend/reparent/utils_test.go

```diff
@@ -0,0 +1,538 @@
+	dbName          = "vt_" + keyspaceName
+	username        = "vt_dba"
+	hostname        = "localhost"
+	insertSQL       = "insert into vt_insert_test(id, msg) values (%d, 'test %d')"
+	sqlSchema       = `
+	create table vt_insert_test (
+	id bigint,
+	msg varchar(64),
+	primary key (id)
+	) Engine=InnoDB	
+`
+)
+
+//region cluster setup/teardown
+func setupRangeBasedCluster(ctx context.Context, t *testing.T) {
+	tablets := setupCluster(ctx, t, shardName, []string{cell1}, []int{2})
+	masterTablet, replicaTablet = tablets[0], tablets[1]
+}
+
+func setupReparentCluster(t *testing.T) {
+	tablets := setupCluster(context.Background(), t, shardName, []string{cell1, cell2}, []int{3, 1})
+	tab1, tab2, tab3, tab4 = tablets[0], tablets[1], tablets[2], tablets[3]
+}
+
+func teardownCluster() {
+	clusterInstance.Teardown()
+}
+
+func setupCluster(ctx context.Context, t *testing.T, shardName string, cells []string, numTablets []int) []*cluster.Vttablet {
+	var tablets []*cluster.Vttablet
+	clusterInstance = cluster.NewCluster(cells[0], hostname)
+	keyspace := &cluster.Keyspace{Name: keyspaceName}
+	// Start topo server
+	err := clusterInstance.StartTopo()
+	if err != nil {
+		t.Fatalf("Error starting topo: %s", err.Error())
+	}
+	err = clusterInstance.TopoProcess.ManageTopoDir("mkdir", "/vitess/"+cells[0])
+	if err != nil {
+		t.Fatalf("Error managing topo: %s", err.Error())
+	}
+	numCell := 1
+	for numCell < len(cells) {
+		err = clusterInstance.VtctlProcess.AddCellInfo(cells[numCell])
+		if err != nil {
+			t.Fatalf("Error managing topo: %s", err.Error())
+		}
+		numCell++
+	}
+
+	// Adding another cell in the same cluster
+	numCell = 0
+	for numCell < len(cells) {
+		i := 0
+		for i < numTablets[numCell] {
+			i++
+			tablet := clusterInstance.NewVttabletInstance("replica", 100*(numCell+1)+i, cells[numCell])
+			tablets = append(tablets, tablet)
+		}
+		numCell++
+	}
+
+	shard := &cluster.Shard{Name: shardName}
+	shard.Vttablets = tablets
+
+	clusterInstance.VtTabletExtraArgs = []string{
+		"-lock_tables_timeout", "5s",
+		"-enable_semi_sync",
+		"-track_schema_versions=true",
+	}
+
+	// Initialize Cluster
+	err = clusterInstance.LaunchCluster(keyspace, []cluster.Shard{*shard})
+	if err != nil {
+		t.Fatalf("Cannot launch cluster: %s", err.Error())
+	}
+
+	//Start MySql
+	var mysqlCtlProcessList []*exec.Cmd
+	for _, shard := range clusterInstance.Keyspaces[0].Shards {
+		for _, tablet := range shard.Vttablets {
+			log.Infof("Starting MySql for tablet %v", tablet.Alias)
+			proc, err := tablet.MysqlctlProcess.StartProcess()
+			if err != nil {
+				t.Fatalf("Error starting start mysql: %s", err.Error())
+			}
+			mysqlCtlProcessList = append(mysqlCtlProcessList, proc)
+		}
+	}
+
+	// Wait for mysql processes to start
+	for _, proc := range mysqlCtlProcessList {
+		if err := proc.Wait(); err != nil {
+			t.Fatalf("Error starting mysql: %s", err.Error())
+		}
+	}
+
+	// We do not need semiSync for this test case.
+	clusterInstance.EnableSemiSync = false
```

Comment:
VTTabletExtraArgs has `-enable_semi_sync`, whereas here it is false. Which one supersedes? Test setup should be unambiguous about it.

Reply:
Not sure what that line did, so I left it there. Removed it and the tests still run, so it seems to have been superfluous code ...